### ISSUE TITLE
transmute_undefined_repr: False positive on tuples of size 1

### ISSUE THREAD

**Issue by dtolnay** [rust-lang/rust-clippy#8503]
### Summary

I believe `transmute_undefined_repr` intentionally avoids linting on structs / tuple structs containing 1 field, but it currently *does* lint on **tuples** containing 1 field. These should be treated consistently with 1-field tuple structs: either lint both or neither.

### Lint Name

transmute_undefined_repr

### Reproducer

```rust
fn main() {
    let _ = unsafe { std::mem::transmute::<&i32, &(i32,)>(&0) };
}
```

```console
$ cargo clippy

error: transmute into `&(i32,)` which has an undefined layout
 --> src/main.rs:2:22
  |
2 |     let _ = unsafe { std::mem::transmute::<&i32, &(i32,)>(&0) };
  |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: `#[deny(clippy::transmute_undefined_repr)]` on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#transmute_undefined_repr
```

```rust
fn main() {
    struct Tuple1(i32);
    let _ = unsafe { std::mem::transmute::<&i32, &Tuple1>(&0) };
}
```

```console
$ cargo clippy
$ # (no lint)
```

### Version

```text
rustc 1.61.0-nightly (8769f4ef2 2022-03-02)
binary: rustc
commit-hash: 8769f4ef2fe1efddd1f072485f97f568e7328f79
commit-date: 2022-03-02
host: x86_64-unknown-linux-gnu
release: 1.61.0-nightly
LLVM version: 14.0.0
```
